src/inventory_ui.cpp

```diff
@@ -846,66 +1261,71 @@ void inventory_compare_selector::toggle_entry( inventory_entry *entry )
-    inventory_selector( u, title, filter ),
-    dropping()
-{
-    insert_selection_column( "ITEMS TO DROP", _( "ITEMS TO DROP" ) );
-}
+inventory_drop_selector::inventory_drop_selector( const player &p,
+        const inventory_selector_preset &preset ) :
+    inventory_multiselector( p, preset, _( "ITEMS TO DROP" ) ) {}
 
 std::list<std::pair<int, int>> inventory_drop_selector::execute()
 {
-    prepare_columns( true );
-
     int count = 0;
     while( true ) {
-        refresh_window();
+        update();
 
-        const std::string action = ctxt.handle_input();
-        const long ch = ctxt.get_raw_input().get_first_input();
-        auto entry = find_entry_by_invlet( ch );
+        const inventory_input input = get_input();
 
-        if( ch >= '0' && ch <= '9' ) {
+        if( input.ch >= '0' && input.ch <= '9' ) {
             count = std::min( count, INT_MAX / 10 - 10 );
             count *= 10;
-            count += ch - '0';
-        } else if( entry != nullptr ) {
-            set_drop_count( *entry, count );
+            count += input.ch - '0';
+        } else if( input.entry != nullptr ) {
+            set_drop_count( *input.entry, count );
             count = 0;
-        } else if( action == "RIGHT" ) {
+        } else if( input.action == "RIGHT" ) {
             for( const auto &entry : get_active_column().get_all_selected() ) {
                 set_drop_count( *entry, count );
             }
             count = 0;
-        } else if( action == "CONFIRM" ) {
+        } else if( input.action == "CONFIRM" ) {
             break;
-        } else if( action == "QUIT" ) {
+        } else if( input.action == "QUIT" ) {
             return std::list<std::pair<int, int> >();
         } else {
-            on_action( action );
+            on_input( input );
             count = 0;
         }
     }
 
     std::list<std::pair<int, int>> dropped_pos_and_qty;
 
     for( auto drop_pair : dropping ) {
-        dropped_pos_and_qty.push_back( std::make_pair( u.get_item_position( drop_pair.first ), drop_pair.second ) );
+        dropped_pos_and_qty.push_back( std::make_pair( p.get_item_position( drop_pair.first ),
+                                       drop_pair.second ) );
     }
 
     return dropped_pos_and_qty;
 }
 
-void inventory_drop_selector::draw( WINDOW *w ) const
+const player &inventory_drop_selector::get_player_for_stats() const
 {
-    inventory_selector::draw( w );
-    // Make copy, remove to be dropped items from that
-    // copy and let the copy recalculate the volume capacity
-    // (can be affected by various traits).
-    player tmp = u;
-    remove_dropping_items(tmp);
-    draw_inv_weight_vol( w, tmp.weight_carried(), tmp.volume_carried(), tmp.volume_capacity() );
-    mvwprintw(w, 1, 0, _("To drop x items, type a number and then the item hotkey."));
+    std::map<item *, int> dummy_dropping;
+
+    dummy.reset( new player( p ) );
```

Comment:
I thought you were in favor of dropping this idiom (duplicating `player` or `npc`)?